.github/workflows/iroha2-dev.yml

```diff
@@ -8,6 +8,26 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  check_format_smart_contracts:
+    runs-on: ubuntu-latest
+    container:
+      image: hyperledger/iroha2-ci:nightly-2023-06-25
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Format check Default Executor
+        run: |
+          cd ./default_executor
+          cargo fmt -- --check
+          cargo check
+
+      - name: Format check Client
+        run: |
+          cd ./client/tests/integration/smartcontracts
+          cargo fmt -- --check
+          cargo check
+
```

Comment:
I think you should put this into `iroha2-dev-pr-static.yml`

Reply:
I think it would be sufficient to just replicate the logic in `iroha2-dev-pr-static.yml`

Reply:
but from what I understood of our github actions so far isn't the static action's funcitonality analysis only?